### PR TITLE
Resolve Conflicts encountered

### PR DESCRIPTION
**Pull request by Nihirraa**

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Allergy.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+/**
+ * Represents an Allergy in the system.
+ * Guarantees: immutable; allergy name is valid as declared in {@link #isValidAllergy(String)}
+ */
+public class Allergy {
+    public static final String MESSAGE_CONSTRAINTS = "ERROR: Invalid allergies format. "
+            + "Allergies should be a non-empty alphanumeric string, and may include "
+            + "commas, spaces, hyphens, or periods.";
+    public static final String VALIDATION_REGEX = "[\\p{Alnum},\\s.-]+";
```
Comment (nayanaamahesh) [inline src/main/java/seedu/address/model/person/Allergy.java:14]: Good use of regex to validate the input.

File: src/main/java/seedu/address/model/person/Allergy.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    public final String allergyName;
+
+    /**
+     * Constructs an {@code Allergy}.
+     *
+     * @param allergyName A valid allergy name.
+     */
+    public Allergy(String allergyName) {
+        requireNonNull(allergyName);
+        allergyName = allergyName.strip();
```
Comment (nayanaamahesh) [inline src/main/java/seedu/address/model/person/Allergy.java:25]: Good that you stripped the allergy name.

File: src/main/java/seedu/address/model/person/Allergy.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        if (other == this) {
+            return true;
+        }
+        // instanceof handles nulls
+        if (!(other instanceof Allergy)) {
+            return false;
+        }
+
+        Allergy otherAllergy = (Allergy) other;
+        return allergyName.equals(otherAllergy.allergyName);
```
Comment (nayanaamahesh) [inline src/main/java/seedu/address/model/person/Allergy.java:48]: Can consider making it case insensitive

File: src/main/java/seedu/address/storage/JsonAdaptedAllergy.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    @JsonValue
+    public String getAllergyName() {
+        return allergyName;
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted allergy object into the model's {@code Allergy} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted allergy.
+     */
```
Comment (Nihirraa) [inline src/main/java/seedu/address/storage/JsonAdaptedAllergy.java:40]: good addition of JavaDoc comments!

File: src/test/java/seedu/address/model/person/AllergyTest.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+import org.junit.jupiter.api.Test;
+
+public class AllergyTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Allergy(null));
+    }
+
+    @Test
```
Comment (Nihirraa) [inline src/test/java/seedu/address/model/person/AllergyTest.java:17]: neatly written with good SLAP

**Pull request merged**